### PR TITLE
process replay: test log comparison

### PR DESCRIPTION
**Pull request by sshane**
Diff coverage was going to changed in https://github.com/commaai/openpilot/pull/30173, so better have a test for this.

### CODE REVIEW

**Comment by sshane** [comment]
@fredyshox made it a fuzzy test which captures the failure + should capture anything else. Let me know which you think is better/more readable.

File: selfdrive/test/process_replay/test_compare_logs.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+class TestCompareLogs(unittest.TestCase):
+  @staticmethod
+  def _msg(which: str):
+    return messaging.new_message(which).as_reader()
+
+  @staticmethod
+  def _get_failed(ref_logs: list, new_logs: list) -> bool:
+    try:
+      diff = compare_logs(ref_logs, new_logs, ignore_fields=IGNORE_FIELDS, ignore_msgs=[], tolerance=None)
```
Comment (sshane) [inline selfdrive/test/process_replay/test_compare_logs.py:20]:
```suggestion
      diff = compare_logs(ref_logs, new_logs, ignore_fields=IGNORE_FIELDS)
```

**Review by fredyshox — APPROVED** [review]
LGTM

File: selfdrive/test/process_replay/test_compare_logs.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    ref_logs = data.draw(st.lists(st.sampled_from(services).map(lambda x: self._msg(x)), max_size=6))
+    new_logs = data.draw(st.lists(st.sampled_from(services).map(lambda x: self._msg(x)), max_size=6))
+
+    any_diff = [remove_ignored_fields(m, IGNORE_FIELDS).as_reader().to_dict(verbose=True) for m in ref_logs] != \
+               [remove_ignored_fields(m, IGNORE_FIELDS).as_reader().to_dict(verbose=True) for m in new_logs]
+
+    try:
+      diff = compare_logs(ref_logs, new_logs, ignore_fields=IGNORE_FIELDS)
+      _, _, failed = format_diff({"": {"": diff}}, {"": {"": {"ref": "", "new": ""}}}, {})
+    except Exception:
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/test_compare_logs.py:32]: this is a really bad pattern and will hide non-test failures

Comment (sshane) [reply selfdrive/test/process_replay/test_compare_logs.py]: made it test for each case explicitly again + tests for exception messages

**Pull request closed without merging**